### ISSUE TITLE
Wrong paths after running ldconfig

### ISSUE THREAD

**Issue by PurpleVsGreen** [zatrazz/rldd#18]
Before:
```
$ rldd -l /bin/sh
        libncursesw.so.6 => /lib64/libncursesw.so.6
        libc.so.6 => /lib64/libc.so.6
        ld-linux-x86-64.so.2 => /lib64/ld-linux-x86-64.so.2
```
After:
```
$ rldd -l /bin/sh 
        libncursesw.so.6 => /lib64/libncursesw.so.6/libncursesw.so.6
        libc.so.6 => /lib64/libc.so.6/libc.so.6
        ld-linux-x86-64.so.2 => /lib64/ld-linux-x86-64.so.2/ld-linux-x86-64.so.2
```

Workaround: `rm /etc/ld.so.cache`

**Comment by zatrazz** [comment]
Is the binary built with DF_1_NODEFLIB (-z,nodefaultlib)? If not, which is the output of ldd on the same binary after running ldconfig? The ld.so.cache is the binary cache used by glibc for symbol resolution.

**Comment by PurpleVsGreen** [comment]
How can I tell? This is the command I used to build it: ``cargo install --path rldd --profile release``

**Comment by zatrazz** [comment]
The command 'readelf -d <binary> | grep -w NODEFLIB' shows if DF_1_NODEFLIB is set. But I think it would be easy to get the output of ldd for the binary (which issues the loader itself) before and after ldconfig (since ldconfig itself change the dynamic library selection). If they match with what rldd is showing, this is the expected behavior.

**Comment by PurpleVsGreen** [comment]
Grep gives no output. The glibc provided ldd appears to work correctly:
```
papaya# ldd /bin/sh
        linux-vdso.so.1 (0x00007fff4c482000)
        libncursesw.so.6 => /usr/lib64/libncursesw.so.6 (0x00007fa5706c5000)
        libc.so.6 => /usr/lib64/libc.so.6 (0x00007fa5704dc000)
        /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007fa57078f000)
papaya# rldd -l /bin/sh
        libncursesw.so.6 => /lib64/libncursesw.so.6
        libc.so.6 => /lib64/libc.so.6
        ld-linux-x86-64.so.2 => /lib64/ld-linux-x86-64.so.2
papaya# ldconfig
papaya# ldd /bin/sh
        linux-vdso.so.1 (0x00007ffd335b4000)
        libncursesw.so.6 => /usr/lib/libncursesw.so.6 (0x00007fdcb3791000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007fdcb35a8000)
        /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007fdcb385d000)
papaya# rldd -l /bin/sh
        libncursesw.so.6 => /usr/lib/libncursesw.so.6/libncursesw.so.6
        libc.so.6 => /usr/lib/libc.so.6/libc.so.6
        ld-linux-x86-64.so.2 => /usr/lib/ld-linux-x86-64.so.2/ld-linux-x86-64.so.2
```

**Comment by zatrazz** [comment]
I see it now. I think this was fixed by 3cee6bf, where -p was printing bogus value. Could you check if main tree works as expected?

**Comment by PurpleVsGreen** [comment]
```
papaya# rm -f /etc/ld.so.cache
papaya# rldd -l /bin/sh
        libncursesw.so.6 => /lib64/libncursesw.so.6
        libc.so.6 => /lib64/libc.so.6
        ld-linux-x86-64.so.2 => /lib64/ld-linux-x86-64.so.2
papaya# ldconfig
papaya# rldd -l /bin/sh
        libncursesw.so.6 => /usr/lib/libncursesw.so.6
        libc.so.6 => /usr/lib/libc.so.6
        ld-linux-x86-64.so.2 => /usr/lib/ld-linux-x86-64.so.2
```

![yay](https://em-content.zobj.net/thumbs/120/mozilla/36/party-popper_1f389.png) ![yay](https://em-content.zobj.net/thumbs/120/mozilla/36/party-popper_1f389.png) ![yay](https://em-content.zobj.net/thumbs/120/mozilla/36/party-popper_1f389.png)

Yay you fix'd it! Thank you so much!
